gitops/snapshot.go

```diff
@@ -235,6 +237,17 @@ func IsSnapshotValid(snapshot *applicationapiv1alpha1.Snapshot) bool {
 	return false
 }
 
+// IsImageDigestValid checks if image url contains digest and it is valid image, return false if check fails
+func IsImageDigestValid(imageUrl string) bool {
```

Comment:
we are hiding what's wrong here, dropping useful information.

I suggest to change this bool function into validation function that returns `nil` when everything is ok and error message with details when validation failed. This error can by used by caller and logged in error log or passed forward

```suggestion
func ValidateImageDigest(imageUrl string) error {
```

Reply:
Actually at first I make this function return `error` when it is not a valid digest, but later I think the requirement says `only adds Components that have digests in their spec.containerImage field to new Snapshots.` and we just `continue` when `containerImage == ""` previously, so I change to just `continue` when the digest is not valid. I think it depends on the customer requirement, do they expect the error is thrown and stop creating snapshot when containerImage is not valid ("" or without Digest)? @dirgim

Reply:
I don't think we should stop creating the Snapshot, but I think we should log (warning/info) why we didn't add the component's image to the new Snapshot.

Reply:
Ok, thank you and @MartinBasti,  let me log some warning for it.

Reply:
Updated.